compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/types/IrTypeUtils.kt

```diff
@@ -108,3 +108,19 @@ fun IrType.getArrayElementType(irBuiltIns: IrBuiltIns): IrType =
         irBuiltIns.primitiveArrayElementTypes[classifier]
             ?: throw AssertionError("Primitive array expected: $classifier")
     }
+
+fun IrType.toArrayType(irBuiltIns: IrBuiltIns): IrType =
+    if (isPrimitiveType()) {
+        irBuiltIns.primitiveArrayForType[this]?.defaultType
+            ?: throw AssertionError("$this not in primitiveArrayForType")
+    } else {
+        IrSimpleTypeBuilder().apply {
+            classifier = irBuiltIns.arrayClass
+            arguments = listOf(
+                IrSimpleTypeBuilder().apply {
+                    classifier = this@toArrayType.classifierOrNull!!
+                    variance = Variance.OUT_VARIANCE
+                }.buildTypeProjection()
```

Comment:
Again, I'm not sure if this has to have `OUT_VARIANCE`.

If it does, the call to `classifierOrNull!!` will fail here unless the extension receiver is an `IrSimpleType`, so why not change the signature to `fun IrSimpleType.toArrayType ...` and use `buildTypeProjection { variance = Variance.OUT_VARIANCE }` here?

Reply:
I saw some cases that have `OUT_VARIANCE`, but didn't dig enough to convince myself it has to be. I'll go with the simplest way you suggested above. Thanks!